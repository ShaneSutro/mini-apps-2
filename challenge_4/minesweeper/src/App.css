.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.cell {
  border: 1px solid grey;
  text-align: center;
  vertical-align: middle;
  height: 40px;
  width: 40px;
}

.cell:hover {
  cursor: crosshair;
}

.flagged-true:hover {
  cursor: not-allowed;
}

.revealed-false {
  background-color: beige;
}

.revealed-true {
  background-color: white;
}