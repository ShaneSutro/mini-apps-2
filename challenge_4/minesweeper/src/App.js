import React, { Component } from 'react';
import { connect } from 'react-redux';

import { testAction } from './actions/testAction';
import { increment } from './actions/increment';

import './App.css';


const mapStateToProps = (state) => ({
  ...state
})

const mapDispatchToProps = (dispatch) => ({
  testAction: () => dispatch(testAction()),
  increment: (number) => dispatch(increment(number))
})

class App extends Component {

  testAction(event) {
    this.props.testAction();
  }

  increment(number) {
    console.log('increment', number)
    this.props.increment(number)
  }

  render() {
    return (
      <div className="App">
        <h1>Oh hey there</h1>

        <pre>
          {
            JSON.stringify(this.props)
          }
        </pre>
        <button onClick={() => this.increment(this.props.testReducer.count)}>Test Action</button>
      </div>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
