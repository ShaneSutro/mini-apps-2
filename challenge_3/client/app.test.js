/* eslint-disable no-undef */
import React from 'react';
import { render, shallow, mount } from 'enzyme';

import App from './app.jsx';

describe('Testing the scoring algorithms', () => {
  test('Should render correctly with no props', () => {
    const app = shallow(<App />);
  });

  test('Should have a Keypad, Frames, and gameOver component', () => {
    const app = render(<App />);
    expect(app.children().length).toBe(3);
  });

  test.only('Should not allow selecting more than 10 in a single frame', () => {
    const app = mount(<App />);
    expect(app.state().frame).toBe(1);
    expect(app.state().scores[0].score.total).toBe(0);
    app.find('#four').simulate('click');
    // app.find('#keypad') // <table>
    //   .childAt(0) // <tbody>
    //   .childAt(0) // <tr>
    //   .childAt(2) // <td> for number 4
    //   .simulate('click');
    app.instance().forceUpdate();
    expect(app.state().frame).toBe(1);
    expect(app.state().scores[0].score.total).toBe(4);
  });
});
