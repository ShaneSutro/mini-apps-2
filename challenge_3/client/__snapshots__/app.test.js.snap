// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the scoring algorithms Should render correctly with no props 1`] = `
<div>
  <h1 />
  <Keypad
    actions={
      Object {
        "select": [Function],
      }
    }
  />
  <Frames
    currentFrame={1}
    frames={
      Object {
        "scores": Array [
          Object {
            "frame": 1,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 2,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 3,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 4,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 5,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 6,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 7,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 8,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 9,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 10,
            "score": Object {
              "one": 0,
              "three": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 11,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
          Object {
            "frame": 12,
            "score": Object {
              "one": 0,
              "total": 0,
              "two": 0,
            },
          },
        ],
      }
    }
    total={0}
  />
</div>
`;
